metpy/calc/basic.py

```diff
@@ -773,6 +773,163 @@ def smooth_n_point(scalar_grid, n=5, passes=1):
+
+    a = lapse rate in NACA standard atmosphere below the isothermal layer 0.0065
+       ^{\circ}C. m.^{-1})
+
+    T_{0} = standard sea-level temperature 288 K
+
+    H_{b} = station elevation in meters (elevation for which station pressure is given)
+
+    n = \frac{a R_{d}}{g} = 0.190284 where R_{d} is the gas constant for dry air
+
+    And solving for p_{mb} results in the equation below, which is used to
+    calculate station pressure (p_{mb})
+
+    .. math:: p_{mb} = \left [A_{mb} ^ n - \left (\frac{p_{0} a H_{b}}{T_0}
+    \right) \right] ^ \frac{1}{n} + 0.3
+
+    """
+    # Make sure the input values have the correct units
+    altim = altimeter_value.to('hPa')
+    elev = height.to('meters')
+
+    # Constant for Mean Sea Level Pressure
+    mslp = 1013.25 * units.hPa
+    # Mean Sea Level Temperature
+    mslt = 288 * units.kelvin
+    # Lapse rate for a standard atmosphere
+    a = 0.0065 * (units.delta_degC / units.m)
+    # N-Value
+    n = (mpconsts.Rd * a / mpconsts.g).to_base_units()
+
+    return (altim ** n - ((mslp ** n * a * elev) / mslt)) ** (1 / n) + (0.3 * units.hPa)
+
+
+@exporter.export
+@preprocess_xarray
+def altimeter_to_sea_level_pressure(altimeter_value, height, temperature):
+    r"""Convert the altimeter setting to sea level pressure.
+
+    This function is useful for working with METARs since most will provide
+    altimeter values, but not sea level pressure, which is often plotted
+    on surface maps. The following definitions of altimeter setting, station pressure, and
+    sea level pressure are taken from the Federal Meteorology Handbook (2013)
+    p.11-1. Altimeter setting is the pressure value to which an aircraft altimeter scale
+    is set so that it will indicate the altitude above mean sea level of an aircraft
+    on the ground at the location for which the value is determined. It assumes a standard
+    atmosphere. Station pressure is the atmospheric pressure at the designated station
+    elevation. Sea-level pressure is a pressure value obtained by the theoretical reduction
+    of barometric pressure to sea level. It is assumed that atmosphere extends to sea level
+    below the station and that the properties of the atmosphere are related to conditions
+    observed at the station. This value is recorded by some surface observation stations,
+    but not all. If the value is recorded, it can be found in the remarks section.Finding
+    the sea level pressure is helpful for plotting purposes and different calculations.
+
+    Parameters
+    ----------
+    altimeter_value : 'pint.Quantity'
+            The altimeter setting value is defined by the METAR or other observation,
+            with units of inches of mercury (in Hg) or millibars (hPa)
+    height  : 'pint.Quantity'
+            Elevation of the station measuring pressure. Often times measured in meters
+    temperature     : 'pint.Quantity'
+            Temperature at the station measured in either Celsius or Fahrenheit
```

Comment:
You convert temperature below, so it can be given in any temperature units.
```suggestion
            Temperature at the station
```